lib/nodejs/vwf.js

```diff
@@ -124,25 +130,34 @@ function HandleFileRequest( request, response ) {
 }
 
 // Serve is the top level function for serving requests. It first attempts to 
-// serve the request based on parsing the incoming URL.
+// serve the request using the the lobby, which is an Express application.
+// Lobby errors are handled using finalhandler in the same way that Express would.
+// Otherwise, it attempts to serve the request based on parsing the incoming URL.
 // If that fails, it continues to attempt to serve the request as a 'proxy' request,
 // if that also does not serve anything to the request, then an attempt is made
 // to handle the request as a simple direct request for a file within the public
 // directory structure.
 // If all that fails, serve up a 404 response since the request was not handled.
 function Serve( request, response ) {
-    var handledRequest = HandleParsableRequest( request, response );
-    if ( ! ( handledRequest ) ) {
-        handledRequest = HandleProxyRequest( request, response );
-    }
-    if ( ! ( handledRequest ) ) {
-        handledRequest = HandleFileRequest( request, response );
-    }
-    if ( ! ( handledRequest ) ) {
-        global.log("404 : " + url.parse( request.url ).pathname )
-        serve._404( response, "404.html" );
-
-    }
+    lobby( request, response, function( err ) {
```

Comment:
That's right. Everything goes to the Express app first, which can handle, modify, reject, or ignore the request. Login is handled like you say. A few specific routes for supporting login/logout and the lobby are handled there. Everything else is ignored and comes back here to `HandleParsableRequest`, etc. and is handled like before.